space_invaders.py

```diff
@@ -1 +1,295 @@
+import sys, pygame, random
+from pygame import K_LEFT, K_RIGHT, K_UP, K_DOWN
 
+pygame.init()
+
+DELAY = 100 # delay between graphic refresh cycles in ms
+ALIEN_WINGS_DOWN_IMAGE_FILE = "alien.bmp"
+ALIEN_WINGS_UP_IMAGE_FILE = "alien_wings_up.png"
+PLAYER_IMAGE_FILE = "cannon.jpg"
+LASER_IMAGE_FILE = "laser.jpg"
+
+SCREEN_ROWS = 10 # number of rows that aliens descend through
+SIZE = WIDTH, HEIGHT = (600, 400) # screen dimensions in pixels
+ALIEN_ROWS = 3
+ALIENS_PER_ROW = 3
+HORIZONTAL_SPACING = WIDTH / ALIENS_PER_ROW # space between alien fleet
+                                    # columns
+VERTICAL_SPACING = HEIGHT / SCREEN_ROWS # space between alien fleet rows
+
+ALIEN_SPEED = 15
+ALIEN_MOVE_PERIOD = 5 # How many refresh cycles it takes for aliens to be
+                # redrawn
+
+PLAYER_SPEED = 15
+LASER_SPEED = 15
+
+RIGHT = (PLAYER_SPEED, 0) # increments of player movement
+LEFT = (-PLAYER_SPEED, 0) # ...
+DOWN = (0, ALIEN_SPEED) # increment of alien movement
+UP = (0, -LASER_SPEED)
+
+BLACK = (0, 0, 0)
+alien_velocity = [ALIEN_SPEED, 0] # first index is x velocity,
+                        # second is y velocity
+                        # all alien's share this velocity
+
+screen = pygame.display.set_mode(SIZE)
+WINGS_DOWN = pygame.image.load(ALIEN_WINGS_DOWN_IMAGE_FILE).convert()
+WINGS_UP = pygame.image.load(ALIEN_WINGS_UP_IMAGE_FILE).convert()
+alien_image = WINGS_DOWN
+
+class Alien(pygame.sprite.Sprite):
+
+    def __init__(self, x, y):
+        """
+        Initialize alien with coordinates (x, y).
+
+        Parameters:
+        -----------
+        x: positive real number
+            The x coordinate at which to initialize the alien.  Floating point
+            numbers are converted to ints.
+        y: positive real number
+            The y coordinate at which to initialize the alien.  Floating point
+            numbers are converted to ints.
+
+        """
+        pygame.sprite.Sprite.__init__(self)
+        self.image = alien_image
+        self.rect = self.image.get_rect()
+        self.rect = self.rect.move(x-self.rect.left, y-self.rect.top) # move
+                                                    # to (x, y)
+        self.velocity = alien_velocity
+
+    def update(self):
+        """
+        Updates the alien's coordinates.  Reverses the alien's velocity at
+        random or if it hits the edges of the window.
+
+        """
+        self.rect = self.rect.move(self.velocity)
+        if self.rect.left < 0 or self.rect.right > WIDTH: # Alien is outside of
```

Comment:
try to have self documenting code
how about:
if self.outside_window():
and then you wont need the comment.
Actually maybe that's overkill since this is so simple, but the point stands.